### PR TITLE
Add docker health check to integrations

### PR DESCRIPTION
**Pull request by mik-laj**
To constantly respond to container failures, I add a health check to all integrations. Thanks to this, Docker is able to detect the problem earlier and restart the container if necessary.

Information about the state of the containers is also available in `docker ps` command.
Screenshot 2021-01-03 at 00 54 18

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: scripts/ci/docker-compose/backend-mysql.yml [hunk 1]
```diff
@@ -36,6 +38,12 @@ services:
       - ../mysql/conf.d:/etc/mysql/conf.d:ro
       - /dev/urandom:/dev/random   # Required to get non-blocking entropy source
       - mysql-db-volume:/var/lib/mysql
+    healthcheck:
+      test: ["CMD", "mysql", "-h", "localhost", "-P", "3306", "-u", "root", "-e", "SELECT 1"]
+      interval: 10s
+      timeout: 10s
+      retries: 120
+
```
Comment (XD-DENG) [inline scripts/ci/docker-compose/backend-mysql.yml:47]: Is there any specific reason why `restart: always` is not added for `mysql` and `postgres`?

Comment (mik-laj) [reply scripts/ci/docker-compose/backend-mysql.yml:47]: Good point. I missed it. Added.

**Review by XD-DENG — APPROVED** [review]
Overall looks good to me.

But I would like to understand a bit more how the values for `retries` are determined , especially it varies quite much among different cases. For example, for `mysql` it is 120, which may be too big to me.

**Review by potiuk — APPROVED** [review]
Really nice. Much better than the in-container check (though the in-container one should stay as a 'sanity check's)

**Comment by mik-laj** [comment]
> But I would like to understand a bit more how the values for retries are determined , especially it varies quite much among different cases. For example, for mysql it is 120, which may be too big to me.

This value was copied from the internet. All health checks in` integration-*. xml` files have identical configuration.
```
      interval: 5s
      timeout: 30s
      retries: 50
```
interval 5s - breeze is used by users so make it responsive. timeout 30s has no reason whatsoever, but that makes sense to me. retries 50 is is the value copied from (in-container health checks)[https://github.com/apache/airflow/blob/6ef23aff802032e85ec42dabda83907bfd812b2c/scripts/in_container/check_environment.sh#L158-L174]. This value was determined experimentally and so far I have the impression that it works.

**Comment by mik-laj** [comment]
@XD-DENG I updated the PR. Now all healt checks have set `interval` to `5s`  and `timeout` to  `30s`. All integration have set `retries` to `50`. All backend have set `retries` to `5`.

**Comment by XD-DENG** [comment]
> @XD-DENG I updated the PR. Now all healt checks have set `interval` to `5s`  and `timeout` to  `30s`. All integration have set `retries` to `50`. All backend have set `retries` to `5`.
> 
> 

Thanks👍

**Review by TobKed — APPROVED** [review]

**Comment by ashb** [comment]
Does docker do anything with these health checks now? Last time I looked (which was 12+ months ago) Docker itself didn't do anything with these.

**Comment by ashb** [comment]
Oh yes I see in your screenshot it does. Cool.

**Pull request merged**